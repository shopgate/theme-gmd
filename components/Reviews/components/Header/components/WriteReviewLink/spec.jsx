/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import {
  mockedStateWithAll,
  mockedStateProductEmpty,
} from 'Components/Reviews/mock';
import WriteReviewLink from './index';

const mockedStore = configureStore();

/**
 * Creates component with provided store state.
 * @param {Object} mockedState Mocked stage.
 * @return {ReactWrapper}
 */
const createComponent = (mockedState) => {
  const Component = (
    <Provider store={mockedStore(mockedState)}>
      <WriteReviewLink />
    </Provider>
  );
  return mount(Component);
};

describe('<WriteReviewLink>', () => {
  let component = null;

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render when current product is set', () => {
    component = createComponent(mockedStateWithAll);
    expect(component).toMatchSnapshot();
    expect(component.find('span').exists()).toBe(true);
  });

  it('should not render when current product is empty', () => {
    component = createComponent(mockedStateProductEmpty);
    expect(component).toMatchSnapshot();
    expect(component.find('span').exists()).toBe(false);
  });
});
