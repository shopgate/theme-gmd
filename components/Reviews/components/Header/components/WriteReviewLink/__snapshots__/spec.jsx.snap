// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WriteReviewLink> should not render when current product is empty 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(WriteReviewLink)>
    <WriteReviewLink
      dispatch={[Function]}
      productId={null}
    />
  </Connect(WriteReviewLink)>
</Provider>
`;

exports[`<WriteReviewLink> should render when current product is set 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(WriteReviewLink)>
    <WriteReviewLink
      dispatch={[Function]}
      productId="foo"
    >
      <ButtonLink
        href="/item/666f6f/write_review"
        noGap={true}
      >
        <ActionButton
          className=""
          disabled={false}
          flat={true}
          loading={false}
          noGap={true}
          onClick={[Function]}
          rippleClassName=""
          rippleSize={null}
          type="secondary"
          wrapContent={true}
        >
          <div
            className="css-1yv2jt6"
          >
            <RippleButton
              className=""
              disabled={false}
              flat={true}
              onClick={[Function]}
              rippleClassName=""
              rippleSize={null}
              type="secondary"
              wrapContent={true}
            >
              <Button
                className=""
                disabled={false}
                flat={true}
                onClick={[Function]}
                type="secondary"
                wrapContent={false}
              >
                <Button
                  className="css-dh0pju "
                  disabled={false}
                  onClick={[Function]}
                >
                  <button
                    className="css-dh0pju  css-1rn65c0"
                    data-test-id="Button"
                    disabled={false}
                    onClick={[Function]}
                  >
                    <Ripple
                      className="css-8s1fw2 "
                      color="#000"
                      disabled={false}
                      fill={true}
                      onComplete={[Function]}
                      overflow={false}
                      size={null}
                    >
                      <div
                        aria-hidden={true}
                        className="css-8s1fw2 "
                        data-test-id="Ripple"
                        onClick={[Function]}
                        style={null}
                      >
                        <Translate
                          className=""
                          params={Object {}}
                          string="reviews.button_add"
                        >
                          <span
                            className=""
                          >
                            reviews.button_add
                          </span>
                        </Translate>
                      </div>
                    </Ripple>
                  </button>
                </Button>
              </Button>
            </RippleButton>
          </div>
        </ActionButton>
      </ButtonLink>
    </WriteReviewLink>
  </Connect(WriteReviewLink)>
</Provider>
`;
